### PR TITLE
[Feature] Print current view option and null checks

### PR DESCRIPTION
**Pull request by awy3**
Added an option for printing the current view. Need a WebViewer update for this to work because the "renderRect" option doesn't currently work

image

### CODE REVIEW

**Comment by ZhijieZhang** [comment]
Can you open PRs to `master` and `legacy/master` as well? Just want to make sure that this feature won't be missing in the newer version.

File: src/components/PrintModal/PrintModal.js [hunk 1]
```diff
@@ -52,10 +53,18 @@ class PrintModal extends React.PureComponent {
       isWatermarkModalVisible: false,
       watermarkModalOption: null,
       existingWatermarks: null,
+      selectedPrintOption: this.allPages,
```
Comment (ZhijieZhang) [inline src/components/PrintModal/PrintModal.js:56]: I think my question regarding this PR is about this change. I'm actually not sure if it's a common or good practice to store a React ref on a React state. Can you explain a little bit more about why we are doing this? Specifically, what problem did you find in the previous implementation, and how does the implementation here resolve that problem?

The `Input` component can be a controlled or uncontrolled component depending on the prop usage. If I'm understanding correctly, in this PR we are trying to make the `Input`a controlled component. That is, the PrintModal component has the full control of whether a print option input is checked or not. The `Input` component in the previous implementation is uncontrolled, and we just listen to the onChange event. I guess I'm not clear about the benefits of making them be controlled?

File: src/components/PrintModal/PrintModal.js [hunk 2]
```diff
@@ -610,6 +663,10 @@ const mapStateToProps = state => ({
   sortStrategy: selectors.getSortStrategy(state),
   colorMap: selectors.getColorMap(state),
   layoutMode: selectors.getDisplayMode(state),
+  printAllPageDisabled: selectors.isElementDisabled(state, 'allPagesPrintOption'),
+  printCurrentPageDisabled: selectors.isElementDisabled(state, 'currentPagePrintOption'),
+  printViewPageDisabled: selectors.isElementDisabled(state, 'currentViewPrintOption'),
+  printCustomPageDisabled: selectors.isElementDisabled(state, 'customPagesPrintOption'),
```
[the remark this answers is not in the document]
Comment (awy3) [reply src/components/PrintModal/PrintModal.js:669]: this is so if the user does "disableElements", the default option won't point to a option that is hidden

File: src/components/PrintModal/PrintModal.js [hunk 3]
```diff
@@ -610,6 +658,7 @@ const mapStateToProps = state => ({
   sortStrategy: selectors.getSortStrategy(state),
   colorMap: selectors.getColorMap(state),
   layoutMode: selectors.getDisplayMode(state),
+  disabledPrintOptionCount: selectors.getDisabledElementsCountFromList(state, ['allPagesPrintOption', 'currentPagePrintOption', 'currentViewPrintOption', 'customPagesPrintOption']),
```
Comment (ZhijieZhang) [inline src/components/PrintModal/PrintModal.js:661]: Why do we need this?

File: src/components/PrintModal/PrintModal.js [hunk 4]
```diff
@@ -56,6 +58,17 @@ class PrintModal extends React.PureComponent {
   }
 
   componentDidUpdate(prevProps) {
+    const printOptions = [this.allPages, this.currentPage, this.currentView, this.customPages];
+    const selectedPrintOption = printOptions.find(printOption => printOption.current?.checked);
+
+    if(!selectedPrintOption){
```
Comment (ZhijieZhang) [inline src/components/PrintModal/PrintModal.js:64]: Can you explain why we need this?

**Comment by awy3** [comment]
Customer no longer needed this

**Pull request closed without merging**